src/node/blockstorage.cpp

```diff
@@ -260,8 +260,11 @@ bool BlockManager::LoadBlockIndex()
     std::sort(vSortedByHeight.begin(), vSortedByHeight.end(),
               CBlockIndexHeightOnlyComparator());
 
+    CBlockIndex* previous_index{nullptr};
     for (CBlockIndex* pindex : vSortedByHeight) {
         if (ShutdownRequested()) return false;
+        if (previous_index && pindex->nHeight > previous_index->nHeight + 1) return false; // corruption, non-contiguous block indexes
```

Comment:
Wouldn't it be better to check for `!=` instead for `>`?

Reply:
I think that would be incorrect, because there will be multiple indexes for a given height in case of stale blocks / forks.

Reply:
Actually it can't - it's sorted. But still the indices coudl be the same.